pkg_resources/__init__.py

```diff
@@ -99,7 +123,20 @@
     stacklevel=2,
 )
 
+if TYPE_CHECKING:
+    from _typeshed import StrPath
+    from types import _LoaderProtocol
```

Comment:
That comes from typeshed. It's private to help prevent accidental runtime imports because typeshed didn't start using `@type_check_only` until very recently (and I believe we'll keep these names private anyway). In other words, `_LoaderProtocol` doesn't exist in the `types` module.

We could potentially ask to move `_LoaderProtocol` from `types.pyi` to `typeshed/__init__.pyi` and see what other typeshed maintainers think.

Reply:
If `_LoaderProtocol` is undocumented, I think we should not import it.

If `typeshed` maintainers are happy to make an utility definition of `LoaderProtocol` in [`_typeshed`](https://github.com/python/typeshed/tree/main/stdlib/_typeshed) accessible for public usage that would be great. Otherwise if the protocol is small, declaring the protocol inline should also be fine.

(Brain dump: probably having an `_typeshed.importlib` module that helps to address the concerns in https://github.com/python/typeshed/issues/11882#issuecomment-2104253881 would be even greater).